main.js

```diff
@@ -0,0 +1,227 @@
+    for (let cellDiv of gameCells) {
+      //Loop of all the cells
+      cellDiv.addEventListener("click", (event) => {
+        if (event.target.innerText === "×" || event.target.innerText === "○") {
+          //Checks if the div has a x or o and do nothing.
+          return;
+        }
+
+        if (nextPlay) {
+          if (nextPlay === true) {
+            event.target.innerText = playerButton;
+            if (playerButton === "×") {
+              statusDiv.innerHTML = "○ is next";
+            } else if (playerButton === "○") {
+              statusDiv.innerHTML = "× is next";
+            }
+
+            nextPlay = !nextPlay;
+          }
+        } else {
+          if (nextPlay === false) {
+            event.target.innerText = oponentButton;
+            if (oponentButton === "○") {
+              statusDiv.innerHTML = "× is next";
+            } else if (oponentButton === "×") {
+              statusDiv.innerHTML = "○ is next";
+            }
+
+            nextPlay = !nextPlay;
+          }
+        }
+        checkWin();
+      });
+    }
+  } else if (gameStart === false) {
+    alert("NOOO Thats IMPOSSIBLEE!!!\nPlease Chose  X or O !");
+  }
+}
+
+//--------------------------------------- Buttons Events ---------------------------\\
+//Start Button to start the game
+startButton.addEventListener("click", countDown);
+function countDown() {
+  startButton.innerText = "Starting in...";
+  timer = setInterval(count, 1000);
+}
+
+//Event listener of vs Player vs Computer
+playerX.addEventListener("click", (evt) => {
+  playMode = "player";
+  if (playMode === "player") {
+    playerX.innerText = " ";
+    playerX1.innerText = " ";
+  }
+});
+playerX1.addEventListener("click", () => {
+  playMode = "computer";
+  if (playMode === "computer") {
+    playerX.innerText = "";
+    playerX1.innerText = "";
+  }
+});
+
+// Event Reset Button
+resetButton.addEventListener("click", () => {
+  for (let cellDiv of gameCells) {
+    cellDiv.innerText = "";
+    cellDiv.innerText = "";
+  }
+  nextTurn();
+
+  clock();
+});
+
+//Set the player as X and oponent as O
+playX.addEventListener("click", () => {
+  playerButton = "×";
+  oponentButton = "○";
+  gameStart = true;
+  playX.innerText = "";
+  playO.innerText = "";
+});
+
+//Set the player as O and oponent as X
+playO.addEventListener("click", () => {
+  playerButton = "○";
+  oponentButton = "×";
+  gameStart = true;
+  playX.innerText = "";
+  playO.innerText = "";
+});
+
+//----------------------------------------Utilitys Function-------------------------------\\
+
+//This function is only to start placing x and o again on the game.
+function nextTurn() {
+  nextPlay = true;
+}
+
+//CountDown Function
+function count() {
+  interval += 1;
+  startButton.innerText = interval != 4 ? interval : "GO!";
+  if (interval === 4) {
+    clearInterval(timer);
+    startButton.innerText = "Start";
+    clock();
+    interval = 0;
+    if (playMode === "player") {
+      startGame();
```

Comment:
This count up function to game play works, but is confusing because as the game resets, the game clock doesn't. So if you replay they game you loose 3 seconds